Kernel/Inference.hpp

```diff
@@ -957,6 +965,23 @@ class Inference
   unsigned inductionDepth() const { return _inductionDepth; }
   void setInductionDepth(unsigned d) { _inductionDepth = d; }
 
+  DHSet<unsigned>* inductionInfo() const { return _inductionInfo; }
+  void addToInductionInfo(unsigned e) {
+    if (!_inductionInfo) {
+      _inductionInfo = new DHSet<unsigned>();
+    }
+    _inductionInfo->insert(e);
+  }
+  void removeFromInductionInfo(unsigned e) {
+    if (_inductionInfo) {
```

Comment:
Is it expected that this function will be called also when _inductionInfo was never allocated? If not, I think it would be good to use ASS(_inductionInfo) to catch potential bugs when the caller is expecting that _inductionInfo is present but in reality it is not.

Reply:
Yes, it should be only performed on a subset of values that are queried from the very same inference, so the check makes sense.